src/Runtime/Scripts/cshtml5.js

```diff
@@ -325,6 +334,158 @@ document.errorCallback = function (error, IndexOfNextUnmodifiedJSCallInList) {
 	window.onCallBack.OnCallbackFromJavaScriptError(idWhereErrorCallbackArgsAreStored);
 }
 
+document.rerouteMouseEvents = function (id) {
+    document.onmouseup = function(e) {
+        if(e.doNotReroute == undefined)
+        {
+               var element = document.getElementByIdSafe(id);
+               document.reroute(e, element);
+        }
+    }
+    document.onmouseover = function(e) {
+       if(e.doNotReroute == undefined)
+        {
+               var element = document.getElementByIdSafe(id);
+               document.reroute(e, element);
+        }
+    } 
+    document.onmousedown = function(e) {
+       if(e.doNotReroute == undefined)
+        {
+               var element = document.getElementByIdSafe(id);
+               document.reroute(e, element);
+        }
+    }                       
+     document.onmouseout = function(e) {   
+       if(e.doNotReroute == undefined)
+        {
+               var element = document.getElementByIdSafe(id);
+               document.reroute(e, element);
+        }
+    }                            
+    document.onmousemove = function(e) {
+       if(e.doNotReroute == undefined)
+        {
+               var element = document.getElementByIdSafe(id);
+               document.reroute(e, element);
+        }
+    }                                    
+    document.onclick = function(e) {   
+       if(e.doNotReroute == undefined)
+        {
+               var element = document.getElementByIdSafe(id);
+               document.reroute(e, element);
+        }
+    }                                     
+    document.oncontextmenu = function(e) {
+       if(e.doNotReroute == undefined)
+        {
+               var element = document.getElementByIdSafe(id);
+               document.reroute(e, element);
+        }
+    }                                      
+    document.ondblclick = function(e) {   
+       if(e.doNotReroute == undefined)
+        {
+               var element = document.getElementByIdSafe(id);
+               document.reroute(e, element);
+        }
+    }
+}
+
+document.setVisualBounds = function(id, left, top, width, height, bSetAbsolutePosition, bSetZeroMargin, bSetZeroPadding) {
+    var element = document.getElementByIdSafe(id);
+    if (element)
+    {
+        element.style.left = left + "px";
+        element.style.top = top + "px";
+        element.style.width = width + "px";
+        element.style.height = height + "px";
+        
+        if (bSetAbsolutePosition) {
+            element.style.position = "absolute";
+        }
+        if (bSetZeroMargin) {
+            element.style.margin = "0";
+        }
+        if (bSetZeroPadding) {
+            element.style.padding = "0";
+        }
+    }
+}
+
+document.setPosition = function(id, left, top, bSetAbsolutePosition, bSetZeroMargin, bSetZeroPadding) {
+    var element = document.getElementByIdSafe(id);

```

Comment:
do not use 'getElementByIdSafe' in js scripts.
```suggestion
    var element = document.getElementById(id);
```